lib.cpp

```diff
@@ -19,6 +19,13 @@ void change_stack_state(const binary_int_function &operation, stack_type &stack)
   stack.push(operation(op_a, op_b));
 }
 
+std::vector<std::string> splitStringBySpace(const std::string& input) {
```

Comment:
i am not sure if this function is really what you need. you will have to read the whole input into one string, although the std::cin stream already works with words so you will have to combine them, just to split them into a vector again.

Reply:
i use an array to evaluate and pushing every value (operators or number) on the stack separately - isn't that the desired function? 
Example 345 * 2 + 3241 - 123
123
-
3241
+
2
*
345

Reply:
I merged your PR already so you can continue using that function, but it might become the case that it remains unused. you will see what i mean as soon as you start reading from standard input...